### PR TITLE
Bael 8407 spring ai openai chat client

### PR DESCRIPTION
**Pull request by etrandafir93**

### CODE REVIEW

File: spring-ai/src/main/java/com/baeldung/springai/chatclient/rest/BlogsController.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @GetMapping("v3")
+    List<Article> askForRecommendationFromContext(@RequestParam(name = "question") String question) {
+        return chatClient.prompt()
+            .advisors(new QuestionAnswerAdvisor(vectorStore, SearchRequest.defaults()))
+            .user(question)
+            .call()
+            .entity(new ParameterizedTypeReference<List<Article>>() {
+
```
Comment (vinipx) [inline spring-ai/src/main/java/com/baeldung/springai/chatclient/rest/BlogsController.java:73]: I don't think we need this empty line here : )

File: spring-ai/src/main/java/com/baeldung/springai/chatclient/rest/BlogsController.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @GetMapping("v2")
+    List<Article> askForArticleRecommendation(@RequestParam(name = "question") String question) {
+        return chatClient.prompt()
+            .user(question)
+            .call()
+            .entity(new ParameterizedTypeReference<List<Article>>() {
+
```
Comment (vinipx) [inline spring-ai/src/main/java/com/baeldung/springai/chatclient/rest/BlogsController.java:62]: I don't think we need this empty line here : )

**Review by vinipx — COMMENTED** [review]
maybe some tests?

File: spring-ai/src/main/java/com/baeldung/springai/chatclient/rest/BlogsController.java (hunk 1 above)
Comment (etrandafir93) [reply spring-ai/src/main/java/com/baeldung/springai/chatclient/rest/BlogsController.java]: I have removed the newlines now, they were added by the formatted 👍

**Comment by etrandafir93** [comment]
> maybe some tests?

@vinipx - I have replied in Jira about the test: it won't be very simple and I don't think they will bring much value here, let's discuss! :)

**Pull request merged**